src/gui/properties/trackerlist.cpp

```diff
@@ -394,22 +394,28 @@ void TrackerList::editSelectedTracker() {
     torrent->replaceTrackers(trackers);
     if (!torrent->isPaused()) {
       torrent->forceReannounce();
-      torrent->forceDHTAnnounce();
     }
 }
 
 #if LIBTORRENT_VERSION_NUM >= 10000
 void TrackerList::reannounceSelected() {
+    QList<QTreeWidgetItem *> selected_items = selectedItems();
+    if (selected_items.isEmpty()) return;
+
     BitTorrent::TorrentHandle *const torrent = properties->getCurrentTorrent();
     if (!torrent) return;
+    QList<BitTorrent::TrackerEntry> trackers = torrent->trackers();
 
-    QList<QTreeWidgetItem *> selected_items = getSelectedTrackerItems();
-    if (selected_items.isEmpty()) return;
+    foreach (QTreeWidgetItem* item, selected_items) {
+      // DHT case
+      if (item->text(COL_TIER).isEmpty() && item->text(COL_URL).contains("DHT")) {
```

Comment:
I tend to do the finding by not relying on the order of items, because it could also work if sorting (by header) is enabled.